src/common/socket.h

```diff
@@ -62,6 +65,25 @@ inline std::string GetHostName() {
   return std::string(buf.c_str());
 }
 
+/*!
+ * \brief ValidateIP validates an ip address.
+ * \param ip The ip address in string format localhost or x.x.x.x format
+ * \return result of operation.
+ */
+inline bool ValidateIP(std::string ip) {
+  if (ip == "localhost") {
+    return true;
+  }
+  std::vector<std::string> list = Split(ip, '.');
+  if (list.size() != 4)
```

Comment:
google c style, enclose with {} if it contains multiple lines

Reply:
Ideally, we should just use std::istringstream to do the validation(to reduce memory complexity). Given that this is a util function, we would like to apply such higher standard